src/Build/Evaluation/ProjectParser.cs

```diff
@@ -262,40 +236,32 @@ private void ParseProjectRootElementChildren(XmlElementWithLocation element)
         /// </summary>
         private ProjectPropertyGroupElement ParseProjectPropertyGroupElement(XmlElementWithLocation element, ProjectElementContainer parent)
         {
-            ProjectXmlUtilities.VerifyThrowProjectAttributes(element, s_validAttributesOnlyConditionAndLabel);
+            ProjectXmlUtilities.VerifyThrowProjectAttributes(element, ValidAttributesOnlyConditionAndLabel);
 
             ProjectPropertyGroupElement propertyGroup = new ProjectPropertyGroupElement(element, parent, _project);
 
             foreach (XmlElementWithLocation childElement in ProjectXmlUtilities.GetVerifyThrowProjectChildElements(element))
             {
-                ProjectPropertyElement property = ParseProjectPropertyElement(childElement, propertyGroup);
+                ProjectXmlUtilities.VerifyThrowProjectAttributes(childElement, ValidAttributesOnlyConditionAndLabel);
+                XmlUtilities.VerifyThrowProjectValidElementName(childElement);
+                ProjectErrorUtilities.VerifyThrowInvalidProject(!XMakeElements.ReservedItemNames.Contains(childElement.Name) && !ReservedPropertyNames.IsReservedProperty(childElement.Name), childElement.Location, "CannotModifyReservedProperty", childElement.Name);
+
+                // All children inside a property are ignored, since they are only part of its value
+                ProjectPropertyElement property = new ProjectPropertyElement(childElement, propertyGroup, _project);
 
                 propertyGroup.AppendParentedChildNoChecks(property);
             }
 
             return propertyGroup;
         }
 
-        /// <summary>
-        /// Parse a ProjectPropertyElement from the element
-        /// </summary>
-        private ProjectPropertyElement ParseProjectPropertyElement(XmlElementWithLocation element, ProjectPropertyGroupElement parent)
```

Comment:
I wouldn't inline the parse methods for each element type, as they offer some conceptual modularization.

Reply:
And the compiler will hopefully inline it. Maybe you could add `[MethodImpl(MethodImplOptions.AggressiveInlining)]` to the methods instead?

Reply:
What I found when I was going through it was a lot of bouncing back and forth.  I found it hard to clean up with so much of the logic spread all over the place.  Some of the methods are part of parsing the children of an element so I thought it made sense to reduce complexity.